server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java

```diff
@@ -58,16 +58,46 @@
     @Nested
     class SingleConsumerTest {
 
+        private ConnectionFactory connectionFactory;
+        private Connection connection;
+        private Channel channel;
+
+        @BeforeEach
+        void setup(DockerRabbitMQ rabbitMQ) throws IOException, TimeoutException {
+            connectionFactory = rabbitMQ.connectionFactory();
+            connection = connectionFactory.newConnection();
+            channel = connection.createChannel();
+        }
+
+        @AfterEach
+        void tearDown() {
+            closeQuietly(connection, channel);
+        }
+
+        @Test
+        void publishedEventWithoutSubscriberShouldNotBeLost() throws Exception {
+            String queueName = createQueue(channel);
+            publishAMessage(channel);
+            awaitAtMostOneMinute.until(() -> messageReceived(channel, queueName));
+        }
+
         @Test
-        void publishedEventWithoutSubscriberShouldNotBeLost(DockerRabbitMQ rabbitMQ) throws Exception {
-            ConnectionFactory connectionFactory = rabbitMQ.connectionFactory();
-            try (Connection connection = connectionFactory.newConnection();
-                 Channel channel = connection.createChannel()) {
-                String queueName = createQueue(channel);
+        void demonstrateDurability(DockerRabbitMQ rabbitMQ) throws Exception {
+            String queueName = createQueue(channel);
+            publishAMessage(channel);
 
-                publishAMessage(channel);
+            rabbitMQ.restart();
 
-                awaitAtMostOneMinute.until(() -> messageReceived(channel, queueName));
+            awaitAtMostOneMinute.until(() -> containerIsRestarted(rabbitMQ));
```

Comment:
that's strange, if the container is restarted we should have lost persisted data, no?

Reply:
docker restart keep filesystem state